src/main/java/com/newts/newtapp/api/application/user/Edit.java

```diff
@@ -0,0 +1,68 @@
+package com.newts.newtapp.api.application.user;
+
+import com.newts.newtapp.api.application.*;
+import com.newts.newtapp.api.errors.*;
+import com.newts.newtapp.entities.User;
+import com.newts.newtapp.api.gateways.UserRepository;
+
+import java.util.ArrayList;
+
+/**
+ * UserInteractor that edits user information.
+ * RequestModel must provide a User ids, the first is the user who wants to follow the other.
+ */
+public class Edit extends UserInteractor<Void, Exception> {
+
+    /**
+     * Initialize a new Edit interactor with given UserRepository.
+     * @param repository    UserRepository to access user data by
+     */
+    public Edit(UserRepository repository) {
+        super(repository);
+    }
+
+    /**
+     * Accepts a Edit request.
+     * @param request   a request stored as a RequestModel
+     */
+    @Override
+    public Void request(RequestModel request) throws UserNotFound, InvalidUsername, InvalidPassword, AlreadyFollowingUser {
```

Comment:
I like your idea here. I think this is too limiting though. What if a user wants to change multiple things? That would require multiple calls to this interactor, which is unnecessary. I think you should have this interactor's requestmodel contain the attributes of UserProfile, that way a user can change any or all of them in one go. You just take their input and set their user details to match, leave the rest to the caller. Please DM me if you have questions about what I mean!